src/repr/src/fixed_length.rs

```diff
@@ -112,6 +140,70 @@ impl FromRowByTypes for Row {
     }
 }
 
+impl IntoRowByTypes for () {
+    /// Empty iterator for unit.
+    type Iter<'a> = Empty<Datum<'a>>;
+
+    /// Returns the `Row` buffer, which is assumed to be empty, without touching it.
+    ///
+    /// This implementation panics if `types` other than `Some(&[])` are provided. This is because
+    /// unit values need to have an empty schema.
+    fn into_row<'a>(&'a self, row_buf: &'a mut Row, types: Option<&[ColumnType]>) -> &'a Row {
+        let Some(&[]) = types else {
+            panic!("Non-empty schema with unit values")
+        };
```

Comment:
A potential improvement here would be to move the type validation elsewhere so it's not on the fast path. Also fine to leave as follow-up work because it's not easy to implement on a trait-abstraction.

Please `#[inline]` the functions, since they're used across crate boundaries.

Reply:
Captured as follow-up work in issue https://github.com/MaterializeInc/materialize/issues/22349. I revisited the title and added a comment there to make it more general and not only about the late `BytesN`. Also added the `#[inline]` annotations in the code.